src/main/java/com/google/summit/symbols/SummitClassResolver.kt

```diff
@@ -0,0 +1,63 @@
+package com.google.summit.symbols
+
+import com.google.summit.ast.CompilationUnit
+import com.google.summit.ast.Node
+import com.google.summit.ast.declaration.ClassDeclaration
+import com.google.summit.ast.declaration.MethodDeclaration
+import com.google.summit.ast.traversal.Visitor
+
+/**
+ * This class is used to resolve symbols in Apex classes.
+ */
+object SummitClassResolver {
+
+  /**
+   * Given a [listOfAsts], return a Map of all the class definitions (declarations)
+   * found in this AST. The key in the map is the qualified name of the class.
+   */
+  internal fun resolveClassesAndMethods(listOfAsts: List<CompilationUnit>): Map<String, SummitResolver.ClassSymbol> {
+    val classMap = mutableMapOf<String, SummitResolver.ClassSymbol>()
+
+    for (ast in listOfAsts) {
+      val visitor = object : Visitor() {
+        override fun visit(node: Node) {
+          if (node is ClassDeclaration) {
+            val classVisitor = MethodDeclarationCollectorVisitor()
+            node.walkSubtree(classVisitor)
```

Comment:
Also for inner classes: this is potentially for O(n^2) with the depth of nesting.

Some thoughts:
* Single walk of subtree? It seems the cleanest solution. However, if you need to track nesting (to understand which class you are immediately inside), the generic traversal interface may need to be extended to visit twice -- once at the start of a subtree and once at the end -- in order to allow you to push and then pop nesting levels. I dunno... maybe we want this or something like it eventually. Thoughts?
* Alternatively, there is also the `skipBelow` override that could be used to switch from the traversal above the node to the traversal of its subtree. Then every node is only visited once.

Reply:
Well, I think we can either:

1. have MethodDeclarationCollectorVisitor stop recursing when it encounters a class (which would solve both this problem and the one above).
2. delete MethodDeclarationCollectorVisitor and do the method visiting in the (currently-) anonymous visitor itself (which would also solve both problems).

I added a TODO for this as well.